modules.d/95nbd/module-setup.sh

```diff
@@ -1,40 +1,69 @@
-#!/bin/bash
+#!/bin/sh
+# This file is part of dracut.
+# SPDX-License-Identifier: GPL-2.0-or-later
 
-# called by dracut
+# Prerequisite check(s) for module.
 check() {
     local _rootdev
 
-    # if an nbd device is not somewhere in the chain of devices root is
+    # If an nbd device is not somewhere in the chain of devices root is
     # mounted on, fail the hostonly check.
     [[ $hostonly ]] || [[ $mount_needs ]] && {
         _rootdev=$(find_root_block_device)
         [[ -b /dev/block/$_rootdev ]] || return 1
         check_block_and_slaves block_is_nbd "$_rootdev" || return 255
     }
+
+    # If the binary(s) requirements are not fulfilled
+    # return 1 to not include the binary.
     require_binaries nbd-client || return 1
 
-    return 0
+    # Return 255 to only include the module, if another module requires it.
+    return 255
+
 }
 
-# called by dracut
+# Module dependency requirements.
 depends() {
-    # We depend on network modules being loaded
+
+    # This module has external dependency on other module(s).
     echo network rootfs-block
+    # Return 0 to include the dependent module(s) in the initramfs.
+    return 0
 }
 
-# called by dracut
+# Install the required kernel module(s) for the module.
 installkernel() {
     instmods nbd
 }
 
-# called by dracut
+# Install the required file(s) for the module.
 install() {
+
+    # Install the required binary for module
     inst nbd-client
+    # Install the required hook for module
     inst_hook cmdline 90 "$moddir/parse-nbdroot.sh"
+    # Install the required script for module
     inst_script "$moddir/nbdroot.sh" "/sbin/nbdroot"
-    if dracut_module_included "systemd-initrd"; then
+
+    if dracut_module_included "systemd"; then
+        # Install dracuts systemd generator for module
         inst_script "$moddir/nbd-generator.sh" $systemdutildir/system-generators/dracut-nbd-generator
+        # Install the type service unit
+        $systemdsystemunitdir/nbd@.service
+
+        # Install the hosts local user configurations if enabled.
+        if [[ $hostonly ]]; then
+            inst_multiple -H -o \
+                /etc/nbdtab \
+                $systemdsystemconfdir/*.mount \
+                $systemdsystemconfdir/nbd@.service \
+                $systemdsystemconfdir/nbd@.service.d/*.conf
```

Comment:
Err `$systemdsystemconfdir/*.mount` should only be included, if it is needed for the root device, or forced added as a dracut device.

Reply:
We always make the assumption that administrator knows what he's doing when he enables hostonly mode right? So in a sense we expect that the administration follows directions like these [1], then adjusts it for dracut in the environment he plans on using it right or? 
https://libguestfs.org/nbdkit-client.1.html#Easy-mounting-at-boot-time

Reply:
`boot-time` != `initrd`

`hostonly` mode is automatically enabled for e.g. Fedora and RHEL.

Here are some points why we can't blindly copy all `*.mount` in the initrd, which the admin configured:
1. it would need including all necessary kernel modules, necessary for the device
2. it would need proper network configuration in the initrd for network devices
3. the initrd should only prepare the root filesystem
4. if the mount point includes a starting SAN, which takes a long time to boot and which is not needed for the root filesystem, this would slow down stuff a lot needlessly
5. ....